### PR TITLE
fix: replace email index

### PR DESCRIPTION
**Pull request by omBratteng**
https://dailydotdev.slack.com/archives/C03C3JZPRU4/p1722941471628759

### CODE REVIEW

File: src/migration/1722942338711-UserEmailIndex.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class UserEmailIndex1722942338711 implements MigrationInterface {
+  name = 'UserEmailIndex1722942338711'
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE INDEX "IDX_user_loweremail" ON "user" ((lower(email)))`,
```
Comment (vpol) [inline src/migration/1722942338711-UserEmailIndex.ts:8]: create index concurrently

File: src/entity/user/User.ts [hunk 2]
```diff
@@ -66,7 +67,6 @@ export class User {
   name: string;
 
   @Column({ type: 'text', nullable: true })
-  @Index('IDX_user_email')
```
Comment (capJavert) [inline src/entity/user/User.ts:69]: Don't we want to keep one of the regular email indexes, probably this one?

File: src/migration/1722942338711-UserEmailIndex.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    );
+    await queryRunner.query(`DROP INDEX "public"."IDX_user_email"`);
+    await queryRunner.query(`DROP INDEX "public"."user_email_index"`);
```
Comment (idoshamun) [inline src/migration/1722942338711-UserEmailIndex.ts:11]: Should we add `if exists` otherwise it will not work locally?

Comment (omBratteng) [reply src/migration/1722942338711-UserEmailIndex.ts]: added

File: src/migration/1722942338711-UserEmailIndex.ts (hunk 1 above)
Comment (omBratteng) [reply src/migration/1722942338711-UserEmailIndex.ts]: done

File: src/entity/user/User.ts (hunk 2 above)
Comment (omBratteng) [reply src/entity/user/User.ts]: Re-added

**Review by rebelchris — APPROVED** [review]

**Review by idoshamun — APPROVED** [review]

**Pull request merged**